custom/icds_reports/reports/awc_reports.py

```diff
@@ -1097,3 +1097,128 @@ def get_beneficiary_details(case_id, awc_id, selected_month):
                 'y': float(recorded_weight) if row.recorded_height else 0
             })
     return beneficiary
+
+
+@quickcache([
+    'order', 'reversed_order', 'awc_id'
+], timeout=30 * 60)
+def get_awc_report_pregnant(order, reversed_order, awc_id):
+    data = CcsRecordMonthly.objects.filter(
```

Comment:
It looks like this still does not use a view or filter by month. This will return a record for every month of a case's existence currently, and have to iterate over every table instead of just the relevant one. A month filter is necessary here, especially because there is a very limited set of months this data could come from.

Reply:
Hi, `CcsRecordMonthly` is using `ccs_record_monthly` as in `icds_reports/models/aggregate.py`
```
class CcsRecordMonthly(models.Model):
    ...
    class Meta(object):
        managed = False
        db_table = 'ccs_record_monthly'
```
And we can not filter it by month as we want to have all of the records at once form all of the examinations to present besides each other for first, second and third trimester.

Reply:
> Hi, CcsRecordMonthly is using ccs_record_monthly as in icds_reports/models/aggregate.py
Yep, that is a table.  It would be great if it could use a view like the ones in icds_reports/migrations/sql_transforms/database_views because it keeps with the assumption that all db access happens through views, which allows us to tell what columns are being used in an easier way.

> And we can not filter it by month
We cannot filter it by a single month but we can add some kind of month filtering. We know that there will not be more than 9 months worth of these, so can limit to that (or 10 to be extra safe although my understanding is that that is not necessary). That way the db call will not scan over the past 2 years of data, which would be very slow.

Reply:
@dmydlo any update on this

Reply:
@calellowitz I am going to send the changes today.